File: draymed_py/tests/test_code_map_transform.py
from draymed.codes.code_map_transform import get_code_map


class TestCodeMapTransform:
    def test_get_code_map(self) -> None:

        output_codes = get_code_map()

        assert (
            output_codes["accessibility_considerations"]["82971005"]["long"]
            == "Impaired Mobility"
        )

        assert (
            output_codes["management_type"]["D0000007"]["long"] == "Diet and Exercise"
        )

        assert (
            output_codes["ethnicity"]["315635008"]["long"] == "White and Black African"
        )

        assert output_codes["closed_reason"]["D0000029"]["short"] == "movedOutOfArea"

File: draymed_js/tests/test_code_map_transform.ts
import {getCodes} from "../src/codes";

describe('matching descriptive names with code from real data', () => {

    const actualCodes = getCodes();

    test('Short name "dietAndExercise" has long name "Diet and exercise"', () => {
        expect(actualCodes.managementType.dietAndExercise.name).toEqual("Diet and Exercise");
    });

    test('Short name "mixedAfrican" has long name "White and Black African"', () => {
        expect(actualCodes.ethnicity.mixedAfrican.name).toEqual("White and Black African");
    });

    test('Short name "movedOutOfArea" has code "D0000029"', () => {
        expect(actualCodes.closedReason.movedOutOfArea.value).toEqual("D0000029");
    });

    test('Short name "oralMeds" has code "386359008"', () => {
        expect(actualCodes.managementType.oralMeds.value).toEqual("386359008");
    });
});
